Multi-window support

andyduke opened the pull request:
Part of the Android player is rewritten to support the multi-window mode. Now, when you turn on the multi-window mode and changing the size of the player window, the video and sound continues without breaks.

The example application is updated - support for the multi-window is added to the AndroidManifest.

solid-software commented on flutter_vlc_player/android/src/main/java/software/solid/fluttervlcplayer/FlutterVlcPlayer.java:
```diff
@@ -647,4 +659,231 @@ private void log(String message) {
 
+    // --- VLCTextureView
+
+    public class VLCTextureView extends TextureView implements TextureView.SurfaceTextureListener, View.OnLayoutChangeListener, IVLCVout.OnNewVideoLayoutListener {
```
Do you mind moving it to a separate file?

andyduke replied:
Yes, it's reasonable.

solid-software reviewed (changes requested):
Thanks for the PR. I've added a styling suggestion please let me know if you can fix it.

solid-software commented:
@alr2413 @mitchross do you guys want to check this PR?

mitchross commented on flutter_vlc_player/android/src/main/java/software/solid/fluttervlcplayer/VLCTextureView.java:
```diff
@@ -0,0 +1,241 @@
+    }
+
+    public void dispose() {
+        log("************************** dispose");
```
remove this or add a log debug

mitchross commented on flutter_vlc_player/android/src/main/java/software/solid/fluttervlcplayer/VLCTextureView.java:
```diff
@@ -0,0 +1,241 @@
+    public void onSurfaceTextureAvailable(SurfaceTexture surface, int width, int height) {
+        log("************************** onSurfaceTextureAvailable");
+
+        if (mSurfaceTexture == null || mSurfaceTexture.isReleased()) {
```
Its better to close the second } if you are going to start with { 

its alot easier to read open and close brackets.

mitchross commented:
@solid-software  this is fine to merge .

solid-software commented:
Thanks @mitchross! @andyduke do you mind adding changelog for this PR? Basically updating this file https://github.com/solid-software/flutter_vlc_player/blob/master/flutter_vlc_player/CHANGELOG.md and pubspec?

andyduke commented:
> Thanks @mitchross! @andyduke do you mind adding changelog for this PR? Basically updating this file https://github.com/solid-software/flutter_vlc_player/blob/master/flutter_vlc_player/CHANGELOG.md and pubspec?

I changed the version in the pubspec, added the version description to the CHANGELOG, and added to the README file a description of how to configure the Android app to enable multi-window support.

solid-software reviewed (approved):
LGTM - thanks @andyduke !
